### ISSUE TITLE
How to copy the images from GPU to CPU more efficiently?

### ISSUE THREAD

**Issue by Jeremy-Yung** [NVIDIA/DALI#1291]
I want to feed the images to keras with dali's preprocessing pipeline. So I have to transform the gpu tensor to array at first. The code is:
```
x, y = pipe.run()
x = x.as_cpu().as_array()
y = y.as_cpu().as_array()
```
But it is very slow and about 200-300 images per second. Is there a faster way?
Thanks!

**Comment by awolant** [comment]
Hi, thanks for the question.
Keras support was mentioned in #1260 
Maybe you can try to copy the data straight to the GPU to be used by Keras? I did small example on how to do something similar [here](https://github.com/NVIDIA/DALI/issues/1196#issuecomment-527396223).

**Comment by awolant** [comment]
Also @JanuszL pointed out it might be worth for you to look in #249 and particularly [this response](https://github.com/NVIDIA/DALI/issues/249#issuecomment-513382969).

**Comment by Jeremy-Yung** [comment]
@awolant I find I can't serialize an external function, just same a s https://github.com/NVIDIA/DALI/issues/123. Does this problem have been solved?

**Comment by awolant** [comment]
No, unfortunately right now it is impossible to serialize the `ExternalSource`.
